### PR TITLE
Spanish translation of sections 1 and 2

### PR DESCRIPTION
**Pull request by jormaechea**

### CODE REVIEW

**Comment by jormaechea** [comment]
@desko27 I translated all the inner content of the first two sessions. I'll await your review..

File: sections/errorhandling/apmproducts.spanish.md [hunk 1]
```diff
@@ -4,25 +4,22 @@
...
 
-3. Operational intelligence dashboard – this line of products is focused on facilitating the ops team with metrics and curated content that helps to easily stay on top of application performance. This usually involves aggregating multiple sources of information (application logs, DB logs, servers log, etc) and upfront dashboard design work. Following are few selected contenders: [Datadog](https://www.datadoghq.com/), [Splunk](https://www.splunk.com/), [Zabbix](https://www.zabbix.com/)
+3. Tableros de inteligencia operativa – esta línea de productos está enfocada en facilitar al equipo de operaciones con métricas y contenido curado que ayude a fácilmente tener visibilidad de la performance de la aplicación. Esto en general involucra agregar múltiples fuentes de información (logs de aplicación, logs de BD, logs de servidores, etc) y trabajo de diseño de tableros por adelantado. Algunos ejemplos son [Datadog](https://www.datadoghq.com/), [Splunk](https://www.splunk.com/), [Zabbix](https://www.zabbix.com/)
```
Comment (desko27) [inline sections/errorhandling/apmproducts.spanish.md:19]:
```suggestion
3. Tableros de inteligencia operativa – esta línea de productos está enfocada en facilitar métricas y contenido curado al equipo de operaciones que ayude a tener visibilidad de la performance de la aplicación con facilidad. Esto, en general, involucra agregar múltiples fuentes de información (logs de aplicación, logs de BD, logs de servidores, etc) y trabajo previo de diseño de tableros. Algunos ejemplos son [Datadog](https://www.datadoghq.com/), [Splunk](https://www.splunk.com/), [Zabbix](https://www.zabbix.com/)
```

File: sections/errorhandling/apmproducts.spanish.md [hunk 2]
```diff
@@ -4,25 +4,22 @@
...
+![alt text](https://github.com/i0natan/nodebestpractices/blob/master/assets/images/uptimerobot.jpg "Dashboard de monitoreo de Sitio web")
 
-
- ### Example: UpTimeRobot.Com – Website monitoring dashboard
-![alt text](https://github.com/i0natan/nodebestpractices/blob/master/assets/images/uptimerobot.jpg "Website monitoring dashboard")
-
- ### Example: AppDynamics.Com – end to end monitoring combined with code instrumentation
-![alt text](https://github.com/i0natan/nodebestpractices/blob/master/assets/images/app-dynamics-dashboard.png "end to end monitoring combined with code instrumentation")
+### Ejemplo: AppDynamics.Com – Monitoreo and to end combinado con instrumentación de código
+![alt text](https://github.com/i0natan/nodebestpractices/blob/master/assets/images/app-dynamics-dashboard.png "Monitoreo and to end combinado con instrumentación de código")
```
Comment (desko27) [inline sections/errorhandling/apmproducts.spanish.md:25]:
```suggestion
### Ejemplo: AppDynamics.Com – Monitoreo end to end combinado con instrumentación de código
![alt text](https://github.com/i0natan/nodebestpractices/blob/master/assets/images/app-dynamics-dashboard.png "Monitoreo end to end combinado con instrumentación de código")
```

File: sections/errorhandling/asyncerrorhandling.spanish.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 Los callbacks no escalan bien ya que muchos programadores no están familiarizados con ellos. Te obligan a estar pendiente de errores por todos lados y a lidiar con un anidamiento de código excesivo, lo cual dificulta la comprensión del flujo de código. Librerías prometedoras como BlueBird, async y Q empaquetan un estilo de código estándar utilizando RETURN y THROW para controlar el flujo del programa. Específicamente, soportan el estilo favorito de gestión de errores try-catch, liberando a la ruta principal de código de tratar con errores en cada función.
 
-### Code Example – using promises to catch errors
+### Código de ejemplo – Uando promesas para capturar errores
```
Comment (desko27) [inline sections/errorhandling/asyncerrorhandling.spanish.md:7]:
```suggestion
### Código de ejemplo – Usando promesas para capturar errores
```

File: sections/errorhandling/shuttingtheprocess.spanish.md [hunk 4]
```diff
@@ -29,23 +29,32 @@ function errorHandler() {
...
+
+ > …Por la misma naturaleza de cómo throw funciona in JavaScript, casi nunca hay una forma segura de "seguir dónde lo dejaste", sin generar fugas de referencias, o sin crear algún otro tipo de estado frágil indefinido. La forma más segura de responder a un error lanzado is apagar el proceso. Claro, en un web server normal, podrías tener muchas conexiones abiertas, y no es rasonable cerrarlas abruptamente porque un error fue disparado por alguna otra persona. El mejor enfoque is enviar una respuesta de error a la petición que disparó el error mientras dejas las otras terminar en su tiempo normal, y parar de escuchar nuevas peticiones en ese worker.
```
Comment (desko27) [inline sections/errorhandling/shuttingtheprocess.spanish.md:60]:
```suggestion
 > …Por la misma naturaleza de cómo throw funciona en JavaScript, casi nunca hay una forma segura de "seguir donde lo dejaste", sin generar fugas de referencias, o sin crear algún otro tipo de estado frágil indefinido. La forma más segura de responder a un error lanzado es apagar el proceso. Claro, en un web server normal, podrías tener muchas conexiones abiertas, y no es razonable cerrarlas abruptamente porque un error haya sido disparado por alguna otra persona. El mejor enfoque es enviar una respuesta de error a la petición que disparó el error mientras dejas las otras terminar en su tiempo normal, y parar de escuchar nuevas peticiones en ese worker.
```

File: sections/errorhandling/testingerrorflows.spanish.md [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 Testear rutas de ejecución "felices" no es mejor que testear errores. Un buen coverage de código exige testear rutas de ejecución adversas. De lo contrario, no existe ninguna seguridad de que las excepciones se capturen y gestionen correctamente. Cada framework de testing unitario, como Mocha y Chai, admite testing de excepciones (códigos de ejemplo a continuación). Si te parece tedioso ir probando cada función y excepción internas, podrías quedarte solo con el testing de errores REST API HTTP.
 
-### Code example: ensuring the right exception is thrown using Mocha & Chai
+### Código de ejemplo: asegurando que la excepción correcta es disparada usando Mocha & Chai
```
Comment (desko27) [inline sections/errorhandling/testingerrorflows.spanish.md:7]:
```suggestion
### Código de ejemplo: asegurando que se dispara la excepción correcta usando Mocha & Chai
```

File: sections/errorhandling/useonlythebuiltinerror.spanish.md [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 La naturaleza permisiva de JS junto con sus variadas opciones de flujo de código (ej. EventEmitter, Callbacks, Promises, etc) provocan la existencia de una gran variedad de maneras de lanzar errores. Algunos utilizan strings, otros definen sus propios tipos personalizados. Utilizar el objeto nativo de Error presente en Node.js ayuda a mantener la uniformidad dentro de tu código y de librerías de terceros, así como a mantener información significativa como el StackTrace. Al lanzar la excepción, normalmente es buena práctica el agregar propiedades adicionales que aporten contexto, como el nombre del error y el código de error HTTP asociado. Para lograr esta uniformidad y estas prácticas, considera extender el objeto Error con propiedades adicionales, observa el ejemplo a continuación.
 
-### Code Example – doing it right
+### Código de ejemplo – Cómo hacerlo bien?
```
Comment (desko27) [inline sections/errorhandling/useonlythebuiltinerror.spanish.md:7]:
```suggestion
### Código de ejemplo – ¿Cómo hacerlo bien?
```

File: sections/errorhandling/useonlythebuiltinerror.spanish.md [hunk 7]
```diff
@@ -24,15 +24,15 @@ return new Promise(function (resolve, reject) {
...
+### Código de ejemplo – Anti patrón
 
 ```javascript
 // throwing a string lacks any stack trace information and other important data properties
 if(!productToAdd)
     throw ("How can I add new product when no value provided?");
 ```
 
-### Code example – doing it even better
+### Código de ejemplo – Cómo hacerlo aún mejor?
```
Comment (desko27) [inline sections/errorhandling/useonlythebuiltinerror.spanish.md:35]:
```suggestion
### Código de ejemplo – ¿Cómo hacerlo aún mejor?
```

**Review by desko27 — APPROVED** [review]
Great! Just some typos and little improvements. Good job and thank you for contributing 👍

**Comment by jormaechea** [comment]
@desko27 Great! Just committed the suggested changed.. RTM!

**Comment by kevynb** [comment]
Great! Thanks @desko27 and @jormaechea 

@all-contributors please add @jormaechea for translation

**Pull request merged**